### ISSUE TITLE
2.4.11 Focus Appearance (Minimum) - the math is difficult to understand and explain 

### ISSUE THREAD

**Issue by CarynPagelTR** [w3c/wcag#1439]
On Behalf of Thomson Reuters 

When our team introduced this Success Criterion to other internal teams, even those of us who are members of the AGWG (accessibility guidelines working group) struggled to explain it in a way that everyone could understand. As it is written now, we are concerned that this Success Criterion will be: 

- Difficult to understand 
- Difficult to test 

A few examples of wording that our teams struggled with: 

- "If a focus indicator has a gradient, the principle is to measure the contrast of the changed area, and ignore the change that does not meet 3:1." Do we have to measure the contrast ratio for many pixels, then perform math equations to determine indication area of gradient that pass contrast minimums? 
- "If the control has a visible boundary smaller than the hit area, the size measure is taken from the visible boundary." Feedback from our internal teams is that focus is applied on the hit area, so using an example where the visible boundary is smaller than the hit area does not make sense. 
- All the talk about pixels had our teams wondering how we would handle elements that use `rem` or `em` or another flexible measurement, rather than pixels. It will require math to convert the element's measurements to pixels, and then determine if this requirement is met. 

By making it difficult to understand and test, we believe this Success Criterion is introducing the possibility of errors from content authors. We understand errors are always possible, but we'd like for the SC requirements to be simple enough to not be the cause of errors through a lack of understanding. 

We propose simplifying the language throughout this Success Criterion and providing more examples with screenshots.

**Comment by alastc** [comment]
Draft response:

<hr>

The group has gone through many rounds of refinement on this criterion, and we think the new version is clearer:
https://w3c.github.io/wcag/guidelines/22/#focus-appearance-minimum

There is a balance though. In defining what a visible focus is, it is very easy to define something that will definitely be visible (e.g. a 2px thick highly contrasting outline), but it would also fail other indicators which are also reasonably visible. 

The complexity comes from what factors make something visible, we have focused it on change of contrast, adjacent contrast, and thickness of the indicator.

On your examples:

> "If a focus indicator has a gradient, the principle is to measure the contrast of the changed area, and ignore the change that does not meet 3:1." Do we have to measure the contrast ratio for many pixels, then perform math equations to determine indication area of gradient that pass contrast minimums?

Similar to the section in the [Non-text contrast understanding doc](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html#figure-blue-circle-i-versions), you would disregard the pixels that don't meet the contrast level. How that is measured depends on the example, but if it is done with something like a box-shadow in CSS, it could be calculated from the definition rather than measuring pixels.

> "If the control has a visible boundary smaller than the hit area, the size measure is taken from the visible boundary." Feedback from our internal teams is that focus is applied on the hit area, so using an example where the visible boundary is smaller than the hit area does not make sense.

In which case it won't be an issue. Feedback from other developers has indicated that sometimes they would increase the hit area outside of the visible boundary.

> All the talk about pixels had our teams wondering how we would handle elements that use rem or em or another flexible measurement, rather than pixels. It will require math to convert the element's measurements to pixels, and then determine if this requirement is met.

All these measures are converted to CSS pixels by the browser to render them, this information is available in the browser inspector. If an outline thickness is defined by EMs/REMs, that will be stable across zoom levels etc. It could vary with text-size settings, but that would then be a user-set preference rather than the default.

> We propose simplifying the language throughout this Success Criterion and providing more examples with screenshots.

We have tried to keep it as simple as possible whilst still providing some flexibility to content authors, and the [AAA version](https://w3c.github.io/wcag/guidelines/22/#focus-appearance-enhanced) is there as an example of a simpler but less flexible version.

The understanding document has 18 screenshots in at the moment, if there is something in particular you think is missing, please do suggest it.

**Comment by alastc** [comment]
As discussed in the meeting, no one has been able to come up with the same sort of requirement in a simpler form. 

We will endeavour to update the understanding document to present easy-to-understand examples right at the top.
